src/Cli/dotnet/commands/dotnet-workload/update/WorkloadUpdateCommand.cs

```diff
@@ -46,71 +49,128 @@ internal class WorkloadUpdateCommand : InstallingWorkloadCommand
+                    }
                 }
-                catch (Exception e)
+                else if (_printDownloadLinkOnly)
                 {
-                    // Don't show entire stack trace
-                    throw new GracefulException(string.Format(LocalizableStrings.WorkloadUpdateFailed, e.Message), e, isUserError: false);
+                    var packageUrls = GetUpdatablePackageUrlsAsync(_includePreviews).GetAwaiter().GetResult();
+                    PrintDownloadLink(packageUrls);
+                }
+                else if (_adManifestOnlyOption)
+                {
+                    var workloadManifestInfo = _workloadResolver.GetInstalledManifests();
+                    recorder.HistoryRecord.WorkloadArguments = workloadManifestInfo.Select(m => m.Id).ToList();
+                    recorder.Run(() => _workloadManifestUpdater.UpdateAdvertisingManifestsAsync(_includePreviews, string.IsNullOrWhiteSpace(_fromCacheOption) ? null : new DirectoryPath(_fromCacheOption), workloadManifestInfo).Wait());
+                    Reporter.WriteLine();
+                    Reporter.WriteLine(LocalizableStrings.WorkloadUpdateAdManifestsSucceeded);
+                }
+                else if (_printRollbackDefinitionOnly)
+                {
+                    var manifestInfo = WorkloadRollbackInfo.FromManifests(_workloadResolver.GetInstalledManifests());
+                    Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputStart==");
+                    Reporter.WriteLine(manifestInfo.ToJson());
+                    Reporter.WriteLine("==workloadRollbackDefinitionJsonOutputEnd==");
+                }
+                else
+                {
+                    recorder.Run(() =>
+                    {
+                        try
+                        {
+                            bool usedRollback = !string.IsNullOrWhiteSpace(_fromRollbackDefinition);
+                            var rollbackFileContents = usedRollback ? WorkloadManifestUpdater.ParseRollbackDefinitionFile(_fromRollbackDefinition, _sdkFeatureBand) : null;
+                            if (usedRollback)
+                            {
+                                var rollbackContents = new Dictionary<string, string>();
+                                recorder.HistoryRecord.WorkloadArguments = new List<string>();
+                                foreach (var (id, version, featureBand) in rollbackFileContents)
+                                {
+                                    var idString = id.ToString();
+                                    recorder.HistoryRecord.WorkloadArguments.Add(idString);
+                                    rollbackContents[idString] = $"{version}/{featureBand}";
+                                }
+
+                                recorder.HistoryRecord.RollbackFileContents = rollbackContents;
+                            }
```

Comment:
It seems like this code may be redundant, as the same thing happens within `_workloadManifestUpdater.CalculateManifestRollbacks`.